src/bin/grin.rs

```diff
@@ -136,14 +138,11 @@ fn real_main() -> i32 {
 		}
 	}
 
+	let (logs_tx, logs_rx) = mpsc::channel::<LogEntry>();
```

Comment:
I have not finished reading this PR fully but posting this here (mainly to remind me to look at this again) - this is an async channel. If the receiver is too slow for any reason then the queue will grow indefinitely.

Reply:
Ok, could try the sync_channel instead with a reasonable buffer? I'm not sure how this would affect the sender (logging) threads if the buffer filled and they were blocked.

Reply:
There's a `try_send` on a sync sender that we can take advantage of here.
Basically just drop log messages on the floor if we cannot safely send it via the buffer.

https://doc.rust-lang.org/std/sync/mpsc/struct.SyncSender.html#method.try_send

Reply:
done this now